- 해당 노트북은 코랩을 통해 작성되었습니다.

### 1. 경로 설정

- 해당 파트는 본인의 경로에 맞게 설정하면 됩니다.

In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [ ]:
data_path = '/content/drive/Shareddrives/KUIAI/data'

- 경로 설정

In [ ]:
path1 = data_path + '/1.건축물 생애이력 관리 데이터(판매시설)'
path2 = data_path + '/2.건축물 생애이력 관리 데이터(제1종근린생활시설)'
path3 = data_path + '/3.건축물 생애이력 관리 데이터(제2종근린생활시설)'

In [ ]:
import os
import pandas as pd

In [ ]:
os.listdir(data_path)

['2.건축물 생애이력 관리 데이터(제1종근린생활시설)',
 '1.건축물 생애이력 관리 데이터(판매시설)',
 '3.건축물 생애이력 관리 데이터(제2종근린생활시설)',
 '상권분석데이터_상권코드별 좌표',
 '서울시 대규모 점포 정보',
 '서울시 우리마을가게 상권분석_매출',
 '서울시 우리마을가게 상권분석_생활인구',
 '서울시 학교정보',
 '서울시 지하철역 도로명주소']

In [ ]:
folder1 = os.listdir(path1)
folder2 = os.listdir(path2)
folder3 = os.listdir(path3)


In [ ]:
folder1

['강동구_BLCM_MAP_20211216.xlsx',
 '강북구_BLCM_MAP_20211216.xlsx',
 '강서구_BLCM_MAP_20211216.xlsx',
 '관악구_BLCM_MAP_20211216.xlsx',
 '광진구_BLCM_MAP_20211216.xlsx',
 '구로구_BLCM_MAP_20211216.xlsx',
 '금천구_BLCM_MAP_20211216.xlsx',
 '노원구_BLCM_MAP_20211216.xlsx',
 '도봉구_BLCM_MAP_20211216.xlsx',
 '동대문구_BLCM_MAP_20211216.xlsx',
 '동작구_BLCM_MAP_20211216.xlsx',
 '마포구_BLCM_MAP_20211216.xlsx',
 '서대문구_BLCM_MAP_20211216.xlsx',
 '서초구_BLCM_MAP_20211216.xlsx',
 '성동구_BLCM_MAP_20211216.xlsx',
 '성북구_BLCM_MAP_20211216.xlsx',
 '송파구_BLCM_MAP_20211216.xlsx',
 '양천구_BLCM_MAP_20211216.xlsx',
 '영등포구_BLCM_MAP_20211216.xlsx',
 '용산구_BLCM_MAP_20211216.xlsx',
 '은평구_BLCM_MAP_20211216.xlsx',
 '종로구_BLCM_MAP_20211216.xlsx',
 '중구_BLCM_MAP_20211216.xlsx',
 '중랑구_BLCM_MAP_20211216.xlsx',
 '강남구_BLCM_MAP_20211209.xlsx']

In [ ]:
folder2

['강남구.xlsx',
 '강동구.xlsx',
 '강북구.xlsx',
 '강서구.xlsx',
 '관악구.xlsx',
 '광진구.xlsx',
 '구로구.xlsx',
 '금천구.xlsx',
 '노원구.xlsx',
 '도봉구.xlsx',
 '동대문구.xlsx',
 '동작구.xlsx',
 '마포구_BLCM_MAP_20210909.xlsx',
 '서대문구_BLCM_MAP_20210909.xlsx',
 '서초구_BLCM_MAP_20210909.xlsx',
 '성동구_BLCM_MAP_20210909.xlsx',
 '성북구_BLCM_MAP_20210909.xlsx',
 '송파구_BLCM_MAP_20210909.xlsx',
 '양천구_BLCM_MAP_20210909.xlsx',
 '영등포구_BLCM_MAP_20210909.xlsx',
 '용산구_BLCM_MAP_20210909.xlsx',
 '은평구_BLCM_MAP_20210909.xlsx',
 '종로구_BLCM_MAP_20210909.xlsx',
 '중구_BLCM_MAP_20210909.xlsx',
 '중랑구_BLCM_MAP_20210909.xlsx']

In [ ]:
folder3

['강동구_BLCM_MAP_20211013.xlsx',
 '강남구_BLCM_MAP_20211013.xlsx',
 '강북구_BLCM_MAP_20211013.xlsx',
 '관악구_BLCM_MAP_20211013.xlsx',
 '강서구_BLCM_MAP_20211013.xlsx',
 '광진구_BLCM_MAP_20211013.xlsx',
 '구로구_BLCM_MAP_20211013.xlsx',
 '금천구_BLCM_MAP_20211013.xlsx',
 '도봉구_BLCM_MAP_20211013.xlsx',
 '노원구_BLCM_MAP_20211013.xlsx',
 '중랑구_BLCM_MAP_20211013.xlsx',
 '동대문구_BLCM_MAP_20211013.xlsx',
 '중구_BLCM_MAP_20211013.xlsx',
 '동작구_BLCM_MAP_20211013.xlsx',
 '마포구_BLCM_MAP_20211013.xlsx',
 '종로구_BLCM_MAP_20211013.xlsx',
 '서대문구_BLCM_MAP_20211013.xlsx',
 '서초구_BLCM_MAP_20211013.xlsx',
 '은평구_BLCM_MAP_20211013.xlsx',
 '성동구_BLCM_MAP_20211013.xlsx',
 '성북구_BLCM_MAP_20211013.xlsx',
 '용산구_BLCM_MAP_20211013.xlsx',
 '송파구_BLCM_MAP_20211013.xlsx',
 '양천구_BLCM_MAP_20211013.xlsx',
 '영등포구_BLCM_MAP_20220110.xlsx']

### 2. 데이터 합치기

In [ ]:
os.chdir(path1)
df1 = pd.DataFrame()
for files in folder1:
    df = pd.read_excel(files, sheet_name=1)
    df1 = pd.concat([df1, df])

os.chdir(path2)
for files in folder2:
    df = pd.read_excel(files, sheet_name=1)
    df1 = pd.concat([df1, df])

os.chdir(path3)
for files in folder3:
    df = pd.read_excel(files, sheet_name=1)
    df1 = pd.concat([df1, df])

In [ ]:
df1.to_csv('data_combine.csv', index=False, encoding='euc-kr')

In [ ]:
df1

NO 일반/집합 구분                     대지위치  ... 친환경인증점수 지능형인증등급 지능형인증점수
0        1       일반       서울특별시 강동구 길동 336-1  ...       0     NaN       0
1        2       일반      서울특별시 강동구 길동 413-45  ...       0     NaN       0
2        3       집합          서울특별시 강동구 길동 54  ...       0     NaN       0
3        4       일반      서울특별시 강동구 둔촌동 435-5  ...       0     NaN       0
4        5       집합        서울특별시 강동구 둔촌동 603  ...       0     NaN       0
...    ...      ...                      ...  ...     ...     ...     ...
2647  2648       일반   서울특별시 영등포구 영등포동8가 83-8  ...       0     NaN       0
2648  2649       일반     서울특별시 영등포구 영등포동8가 84  ...       0     NaN       0
2649  2650       일반  서울특별시 영등포구 영등포동8가 85-11  ...       0     NaN       0
2650  2651       일반  서울특별시 영등포구 영등포동8가 85-12  ...       0     NaN       0
2651  2652       집합     서울특별시 영등포구 영등포동8가 91  ...       0     NaN       0

[120009 rows x 42 columns]

- 총 120009건의 생애이력 데이터가 결합되었습니다.

### 3. 생애이력 데이터 좌표 추출

- 필요 패키지 불러오기

In [ ]:
import requests
import pandas as pd
import numpy as np
import folium
from folium.plugins import MiniMap
import json

- 해당 데이터셋 불러오기

In [ ]:
dataset = pd.read_csv('/Users/anjeongsu/Desktop/해커톤/data_road.csv', encoding='utf-8')
address=dataset['도로명주소']

- 좌표 추출의 과정은 구글 api를 활용하여 추출했습니다.

In [ ]:
pip install -U googlemaps

In [ ]:
import googlemaps

In [ ]:
# 구글 개인 REST APP KEY -> 함부로 노출하면 안되는 정보이기에 텍스트 파일을 만들어서 열기
with open("./GoogleApiKey.txt", "r") as api_key:
    my_key = api_key.read() 
maps = googlemaps.Client(key=my_key)  # my key값 입력

lat = []  #위도
lng = []  #경도

i=0
for place in address:   
    i = i + 1
    try:
        print("%d번 인덱스에서 %s의 위치를 찾고있습니다"%(i, place))
        geo_location = maps.geocode(place)[0].get('geometry')
        lat.append(geo_location['location']['lat'])
        lng.append(geo_location['location']['lng'])
        

    except:
        lat.append('')
        lng.append('')
        print("%d번 인덱스 위치를 찾는데 실패했습니다."%(i))

df = pd.DataFrame({'위도':lat, '경도':lng}, index=address)

df.to_csv('data_road2.csv', encoding='euc-kr' )